src/store/Business/actions.ts

```diff
@@ -11,7 +11,10 @@ export default {
       const businessId = sessionStorage.getItem('BUSINESS_ID')
 
       // if there is no business id, return error
-      if (!businessId) reject(new Error('Missing business id'))
+      if (!businessId) {
+        reject(new Error('Missing business id'))
+        return
+      }
```

Comment:
I tested this, and sometimes code after "reject" is executed, so a return is needed.

Reply:
That's a good find!  Can you share your test so we can all learn the circumstances where a return is required.

Reply:
I don't have a specific test for this -- I added a console.log statement after the reject call and checked if it ran or not.

In fact, in this action, the regular statement after the reject call was not called. However, in loadFilings() (in FHL actions file), the regular statement after the reject call WAS called. So I kept it consistent between both actions.

I just Googled it again and it looks like reject is supposed to act like a return statement. Except I saw above that it sometimes doesn't. To add to the mystery, my code tools tell me when subsequent statements are never run (eg, due to a throw or return) but here they don't say anything.

Thoughts?

Reply:
From memory, I didn't think that anything after a `resolve()` or `reject()` would be executed, but it's always the unknown, unknowns that surprise me.

Reply:
Looking at the docs, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise it looks like a return is not needed, but the resolve or reject logic should be contained, so no further code can be executed.

For example:
```javascript
function promiseGetWord(parityInfo) {
  return new Promise((resolve, reject) => {
    const { value, isOdd } = parityInfo;
    if (value >= THRESHOLD_A - 1) {
      reject(`Still too large: ${value}`);
    } else {
      parityInfo.wordEvenOdd = isOdd ? "odd" : "even";
      resolve(parityInfo);
    }
  });
}
```

Reply:
> From memory, I didn't think that anything after a `resolve()` or `reject()` would be executed, but it's always the unknown, unknowns that surprise me.

That's my expectation, too, but I know what I saw. How about we leave the "extra" return in there just to be safe?

In your example, there is no other code after the reject or resolve, so 🤷‍♂️